internal/provisioner/kops_provisioner_cluster.go

```diff
@@ -804,30 +802,48 @@ func (provisioner *KopsProvisioner) ResizeCluster(cluster *model.Cluster, awsCli
 	logger.Info("Resizing cluster")
 
 	for igName, changeMetadata := range kopsMetadata.GetWorkerNodesResizeChanges() {
-		logger.Infof("Resizing instance group %s to %d nodes", igName, changeMetadata.NodeMinCount)
-
-		igManifest, err := kops.GetInstanceGroupYAML(kopsMetadata.Name, igName)
-		if err != nil {
-			return err
-		}
-
-		igManifest, err = grossKopsReplaceSize(
-			igManifest,
-			kopsMetadata.ChangeRequest.NodeInstanceType,
-			fmt.Sprintf("%d", changeMetadata.NodeMinCount),
-			fmt.Sprintf("%d", changeMetadata.NodeMaxCount),
-		)
-		if err != nil {
-			return errors.Wrap(err, "failed to update instance group yaml file")
+		// There is a bit of complexity with updating min and max instancegroup
```

Comment:
I do not think this logic is tested in anyway, which makes me a bit uncomfortable as it is very easy to screw something up if we ever change something or even refactor. It may be a bit hard to do it given all the dependencies, but in my opinion, might be worth it.

Reply:
Fair enough. Let me see if I can add these quickly.